### PR TITLE
Upgrade bns codec to v1.0.4

### PR DESCRIPTION
**Pull request by davepuchyr**

### CODE REVIEW

File: packages/iov-bns/src/decodeobjects.ts [hunk 1]
```diff
@@ -170,6 +170,7 @@ export function decodeDomain(prefix: IovBech32Prefix, domain: codecImpl.account.
     hasSuperuser: ensure(domain.hasSuperuser, "hasSuperuser"),
     msgFees: ensure(domain.msgFees, "msgFees").map(decodeAccountMsgFee),
     accountRenew: asIntegerNumber(ensure(domain.accountRenew, "accountRenew")),
+    broker: domain.broker ? encodeBnsAddress(prefix, domain.broker) : ("" as Address),
```
Comment (davepuchyr) [inline packages/iov-bns/src/decodeobjects.ts:173]: @albertandrejev, is `("" as Address)` the right value for the case when there's no broker?  If not then what's the right value?

Comment (abefernan) [reply packages/iov-bns/src/decodeobjects.ts:173]: I believe so, the Domain interface does not let broker be optional. The only concern is what bnsd does with and empty string broker but I think it's correct.

File: packages/iov-bns/src/decodemsg.ts [hunk 2]
```diff
@@ -231,7 +231,7 @@ function decodeRegisterAccountTx(
     name: ensure(msg.name, "name"),
     owner: encodeBnsAddress(prefix, ensure(msg.owner, "owner")),
     targets: ensure(msg.targets, "targets").map(decodeBlockchainAddress),
-    broker: encodeBnsAddress(prefix, ensure(msg.thirdPartyToken, "owner")) || undefined,
+    broker: encodeBnsAddress(prefix, ensure(msg.broker, "owner")) || undefined,
```
Comment (abefernan) [inline packages/iov-bns/src/decodemsg.ts:234]: Is it intentional that broker is "owner" in `decodeRegisterAccountTx` but "broker" in `decodeRegisterDomainTx`?.
I understand that `thirdPartyToken` also did this but just wanted to be sure.

Comment (davepuchyr) [reply packages/iov-bns/src/decodemsg.ts:234]: Great catch!!!  That's a bug.  I'll fix it.

Comment (davepuchyr) [reply packages/iov-bns/src/decodemsg.ts:234]: Fixed by https://github.com/iov-one/iov-core/pull/1419

**Pull request merged**